connector/keystone/keystone.go

```diff
@@ -153,48 +170,32 @@ func (c *Config) Open(id string, logger log.Logger) (connector.Connector, error)
 func (p *conn) Close() error { return nil }
 
 func (p *conn) Login(ctx context.Context, scopes connector.Scopes, username, password string) (identity connector.Identity, validPassword bool, err error) {
-	resp, err := p.getTokenResponse(ctx, username, password)
-	if err != nil {
-		return identity, false, fmt.Errorf("keystone: error %v", err)
-	}
-	if resp.StatusCode/100 != 2 {
-		return identity, false, fmt.Errorf("keystone login: error %v", resp.StatusCode)
-	}
-	if resp.StatusCode != 201 {
-		return identity, false, nil
-	}
-	token := resp.Header.Get("X-Subject-Token")
-	data, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return identity, false, err
-	}
-	defer resp.Body.Close()
-	tokenResp := new(tokenResponse)
-	err = json.Unmarshal(data, &tokenResp)
-	if err != nil {
-		return identity, false, fmt.Errorf("keystone: invalid token response: %v", err)
+	var token string
+	var tokenInfo *tokenInfo
+	if username == "" || username == "_TOKEN_" {
```

Comment:
Won't this break our current approach where we encode `{"username": "<>", "password": "<>"}` and passing it as token in the auth section? Did that approach ever work after dex was introduced ?

Reply:
That approach is still present. It did work for regular keystone users, it never worked for Keystone SSO users